libunwind/src/UnwindCursor.hpp

```diff
@@ -2917,7 +2980,8 @@ template <typename A, typename R> int UnwindCursor<A, R>::step(bool stage2) {
 
   // Use unwinding info to modify register set as if function returned.
   int result;
-#if defined(_LIBUNWIND_CHECK_LINUX_SIGRETURN)
+#if defined(_LIBUNWIND_CHECK_LINUX_SIGRETURN) ||                               \
+    (defined(_LIBUNWIND_TARGET_HAIKU) && defined(_LIBUNWIND_TARGET_X86_64))
```

Comment:
Why is `_LIBUNWIND_TARGET_X86_64` check needed?

Reply:
I suppose because signal frame unwinding is only tested on x86_64. 
@X547

Reply:
riscv64 support can be added with a little effort.

Reply:
I think you could remove the _LIBUNWIND_TARGET_X86_64 checks for all but around line 2758.